EIPS/eip-4377.md

```diff
@@ -0,0 +1,360 @@
+---
+eip: 4377
+title: A Protocol for ERC20 Tokens Which Released Based On Predefined Periods Controlled by Smart Contracts
+description: Controlled by smart contracts, tokens can be linearly released based on the predefined period.
+author: Roy Davies(roydj2021@gmail.com), Ali Kaya (ali.kayatur747@gmail.com)
+discussions-to: https://ethereum-magicians.org/t/erc4377-a-protocol-for-erc20-tokens-which-released-based-on-predefined-periods-controlled/19305
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-03-08
+---
+
+## Abstract
+
+A protocol for ERC20 tokens which released based on predefined periods controlled by smart contracts, enhancing the ERC repository. 
+The smart contract allows token released based on predefined periods, enabling lock token balances and gradually release tokens.
+
+Our team has introduced a novel “Sorter” mechanism in this protocol, which automatically consolidates multiple transfers in smart contracts. This feature ensures token release based on different transfer times while minimizing gas fees significantly. 
+Furthermore, the protocol supports customizable features such as:
+
+Customized White List: Specified addresses or contracts are exempted from the above features and locking restrictions, allowing this protocol to be widely and properly applied. For example, Uniswap contracts, treasury addresses.... etc.
+
+## Motivation
+The motivation behind this protocol is to enrich the ERC repository. By allowing ERC20 tokens to have custom linear release schedules, lock-up mechanisms, and whitelisting, this protocol can find applications in scenarios such as MEME tokens, games, DAO organizations, and options trading.
+
+## Specification
+
+This protocol introduces improvements in the following areas:
+
+1. ERC20 Tokens and Linear Release Functionality.
+2. Online Representation of ERC20 Balances.
+3. Lock-Up Functionality for ERC20 Tokens.
+4. "Sorter" Algorithm.
+5. Low-Cost Swap Algorithm.
+
+## Rationale
```

Comment:
This doesn't explain the choices you have made for this EIP, and code block should generally be included in the Reference Implementation. Use this section to help the reader understand why you have done something in plain English.